enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/subscription/SubscriptionDataStoreImpl.java

```diff
@@ -159,14 +169,18 @@ public void addSubscriptions(List<org.wso2.choreo.connect.discovery.subscription
             newSubscription.setAppUUID(subscription.getAppUUID());
             newSubscription.setSubscriptionState(subscription.getSubscriptionState());
             newSubscription.setTimeStamp(subscription.getTimeStamp());
+            newSubscription.setContext(subscription.getContext());
+            newSubscription.setVersionRange(subscription.getVersionRange());
```

Comment:
We can discard the entry, if the getVersionRange property is not there. This ensures that only newly created subscriptions are getting intelligent routing.